probe/probe.go

```diff
@@ -150,8 +173,14 @@ func (p *Probe) publishLoop() {
 	for {
 		select {
 		case <-pubTick:
-			localReport := p.rpt.swap(report.MakeReport())
-			if err := rptPub.Publish(localReport); err != nil {
+			rpt := condense(report.MakeReport(), p.spiedReports)
+			if err := rptPub.Publish(rpt); err != nil {
+				log.Printf("publish: %v", err)
+			}
+
+		case rpt := <-p.shortcutReports:
+			rpt = condense(rpt, p.shortcutReports)
+			if err := rptPub.Publish(rpt); err != nil {
```

Comment:
I considered using one channel - the side effect would be that shortcut reports could get merged with any pending 'full' reports, potentially making the thing we push much larger, more often.  Given we had to back off the publish tick to 3s due to cpu usage previously, I decided not to do that.